test/Constraints/bridging.swift

```diff
@@ -180,7 +180,8 @@ func dictionaryToNSDictionary() {
 // In this case, we should not implicitly convert Dictionary to NSDictionary.
 struct NotEquatable {}
 func notEquatableError(_ d: Dictionary<Int, NotEquatable>) -> Bool {
-  return d == d // expected-error{{referencing operator function '==' on 'Dictionary' requires that 'NotEquatable' conform to 'Equatable'}}
+  // There is a note attached to a requirement `requirement from conditional conformance of 'Dictionary<Int, NotEquatable>' to 'Equatable'`
+  return d == d // expected-error{{operator function '==' requires that 'NotEquatable' conform to 'Equatable'}}
```

Comment:
Why did this error message (and others below) change to remove the `on 'Dictionary'` part? I think the old one is more descriptive, even with the note pointing to the requirement

Reply:
That's because it now picks a different overload - `extension Dictionary: Equatable where Value: Equatable`, and before it used to pick one on `Equatable` directly. It seems like a better choice now even thought it doesn't say `on Dictionary` anymore since underlying issue is a conditional requirement failure.